File: EEDU-Backend/src/main/java/de/gaz/eedu/course/model/CourseCreateModel.java
package de.gaz.eedu.course.model;

import de.gaz.eedu.course.CourseEntity;
import de.gaz.eedu.entity.model.CreationModel;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public record CourseCreateModel(@NotNull String name, @NotNull Long subjectId, @Nullable Long classId, @NotNull Long[] users) implements CreationModel<CourseEntity>
{
    @Override
    public @NotNull CourseEntity toEntity(@NotNull CourseEntity entity)
    {
        entity.setName(name());
        return entity;
    }
}

File: EEDU-Frontend/src/app/user/courses/models/course-create-model.ts
export class CourseCreateModel {

    public constructor(
        private readonly _name: string,
        private readonly _subjectId: number,
        private readonly _classId?: number,
        private readonly _users: number[] = []
    ) {}

    public static fromObject(obj: {
        name: string,
        subjectId: number,
        classId?: number,
        users: number[]
    }): CourseCreateModel {
        return new CourseCreateModel(obj.name, obj.subjectId, obj.classId, obj.users);
    }

    public get toPacket(): { name: string; subjectId: number, classId?: number, users: number[] }
    {
        return {
            name: this.name, subjectId: this.subjectId, classId: this.classId, users: this.users
        }
    }

    public get name(): string {
        return this._name;
    }

    public get subjectId(): number {
        return this._subjectId;
    }

    public get classId(): number | undefined {
        return this._classId;
    }

    public get users(): number[] {
        return this._users;
    }
}
